lib/index.js

```diff
@@ -166,10 +170,18 @@ MapboxGeocoder.prototype = {
     }
   }, 200),
 
+  _showButton: function() {
+    if (this._typeahead.selected) this._clearEl.style.display = 'block';
+  },
```

Comment:
When close button gets added, we should add some extra right padding to the input to shift where truncation begins:
![Screen Shot 2019-03-19 at 11 34 44 AM](https://user-images.githubusercontent.com/108094/54632314-0b065100-4a3b-11e9-867c-c2c519ba33e7.png)